src/solvers/refinement/string_expr.h

```diff
@@ -0,0 +1,97 @@
+/******************************************************************\
+
+Module: String expressions for the string solver
+
+Author: Romain Brenguier, romain.brenguier@diffblue.com
+
+\*******************************************************************/
+
+#ifndef CPROVER_SOLVERS_REFINEMENT_STRING_EXPR_H
+#define CPROVER_SOLVERS_REFINEMENT_STRING_EXPR_H
+
+#include <langapi/language_ui.h>
+
+#include <solvers/refinement/bv_refinement.h>
+#include <solvers/refinement/refined_string_type.h>
+
+
+// Expressions that encode strings
+class string_exprt: public struct_exprt
+{
+public:
+  // Initialize string from the type of characters
+  explicit string_exprt(unsignedbv_typet char_type);
+
+  // Default uses C character type
+  string_exprt() : string_exprt(refined_string_typet::char_type()) {}
+
+  // Generate a new symbol of the given type tp with a prefix
+  static symbol_exprt fresh_symbol
+  (const irep_idt &prefix, const typet &tp=bool_typet());
+
+  // Expression corresponding to the length of the string
+  inline const exprt & length() const
+  { return op0(); }
+
+  // Expression corresponding to the content (array of characters) of the string
+  inline const exprt & content() const
+  { return op1(); }
```

Comment:
As you only have const variants of those functions, you are directly using op0() and op1() in your above code. I think you should introduce the non-const versions and avoid direct access to op0() and op1(), but instead use the API that you've nicely built.